File: Euler_7/Euler_7.swift
import Foundation
//we import Foundation in order to use sqrt(x)

func isPrime(n: Int) -> Bool{
  if n % 2 == 0{
  	return false
  }
  let n_double = Double(n)
  let max = sqrt(n_double)
  var divisor = 3.0

  while divisor <= max{
  	var remain = n_double.truncatingRemainder(dividingBy: divisor)
  	if remain == 0{
  		return false
  	}
  	divisor += 2
  }
  return true
}

var count = 6, check = 14
while check < 200000{
  if isPrime(n:check){
    count += 1
  }
  if count == 10001{
    break
  }
  check += 1
}

print(check)

File: Euler_7/Euler_7.java
class Euler_7{
  public static void main(String[] args) {
    int count = 6;
    int check = 14;

    while (check < 200000){
      if (isPrime(check)){
        count++;
      }
      if(count == 10001){
        break;
      }
      check++;
    }
    System.out.println(check);
  }
  public static boolean isPrime(int n){
    if (n%2 == 0){
      return false;
    }
    double max = Math.sqrt(n);
    double divisor = 3.0;

    while(divisor <= max){
      if (n%divisor == 0){
        return false;
      }
      divisor += 2;
    }
    return true;
  }
}
